packages/suite/src/components/suite/labeling/MetadataLabeling/MetadataLabeling.tsx

```diff
@@ -226,12 +229,14 @@ export const MetadataLabeling = (props: Props) => {
         };
     }, [props.payload.defaultValue, timeout]);
 
-    const isLabelingAvailable = useSelector(selectIsLabelingAvailable);
-
-    // labeling is possible (it is possible to make it available) when we may obtain keys from device. If enabled, we already have them
-    // (and only need to connect provider), or if device is connected, we may initiate TrezorConnect.CipherKeyValue call and get them
-    const labelingPossible =
-        !props.isDisabled && (device?.metadata.status === 'enabled' || device?.connected);
+    const isLabelingInitPossible = useSelector(selectIsLabelingInitPossible);
+    const isLabelingAvailable = useSelector(state =>
+        selectIsLabelingAvailableForEntity(
+            state,
+            props.payload.entityKey,
+            props.payload.type === 'walletLabel' ? props.payload.entityKey : undefined,
+        ),
+    );
```

Comment:
Condition for walletLabel is super confusing and the explanatory comment is only one of the places, would be nice to separate all methods that has to select device either by deviceState or just as selectedDevice...

But as a compromise, I'd suggest at least putting it in constanty here and reuse that constant on line 257 for init action:
```suggestion
    // if this is wallet(device) label, device has to be selected from list of devices by deviceState instead of just taking selectedDevice
    const deviceState = props.payload.type === 'walletLabel' ? props.payload.entityKey : undefined;
    const isLabelingAvailable = useSelector(state =>
        selectIsLabelingAvailableForEntity(
            state,
            props.payload.entityKey,
            deviceState,
        ),
    );
```

Reply:
ok, 202e077d4f4d8bb614ee1800cc23e6dc22f8695e